### PR TITLE
CAT-2327: adds new SetLookByIndexBrick

### PR DESCRIPTION
**Pull request by comeclarity**
adds the functionality to set a look by index (via formula)
same for setBackground and setBackgroundAndWait

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookByIndexActionTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+		ActionFactory factory = sprite.getActionFactory();
+		Action action = factory.createSetLookByIndexAction(sprite, formula);
+		action.act(1.0f);
+		assertEquals("Action didn't set the first LookData", firstLookData, sprite.look.getLookData());
+
+		formula = new Formula(2);
+		action = factory.createSetLookByIndexAction(sprite, formula);
+		action.act(1.0f);
+		assertEquals("Action didn't set the second LookData", secondLookData, sprite.look.getLookData());
```
Comment (cheidenreich) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookByIndexActionTest.java]: we might also want to test formula values outside the looks range here

File: catroid/src/main/java/org/catrobat/catroid/content/actions/SetLookByIndexAction.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+public class SetLookByIndexAction extends SetLookAction {
+
+	private Formula formula;
+
+	@Override
+	protected void doLookUpdate() {
+
+		updateLookFromFormula();
+		if (wait) {
+			BackgroundWaitHandler.addObserver(look, this);
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/SetLookByIndexAction.java]: should this really be done before the check below?
what happens if the observer gets added, but the `setLookData()` never gets called?

**Comment by cheidenreich** [comment]
this introduces new bricks, isn't a language version increment is required?

**Comment by cheidenreich** [comment]
please run espresso tests

**Comment by comeclarity** [comment]
Thanks for the feedback. I updated the pull request.

**Comment by cheidenreich** [comment]
please run espresso tests

**Pull request merged**